src/main/connect/connectionManager.ts

```diff
@@ -130,6 +136,107 @@ export class ConnectionManager implements ExtensionComponent {
+
+        if (prompt) {
+            return await this.promptAll();
+        }
+        return false;
+    }
+
+    private async promptAll(): Promise<boolean> {
+        if (!(await this.promptUrls())) {
+            return false;
+        }
+        if (await this.promptAccessToken()) {
+            return true;
+        }
+        return (await this.promptUsername()) && (await this.promptPassword());
+    }
+
+    public async readCredentialsFromJfrogCli(): Promise<boolean> {
+        if (!(await this.verifyJfrogCliInstalledAndVersion())) {
+            return false;
+        }
+        return await this.getJfrogCliDefaultServerConfiguration();
+    }
+
+    private async verifyJfrogCliInstalledAndVersion(): Promise<boolean> {
+        try {
+            let version: string = await this.getJfrogCliVersion();
+            if (!version) {
+                return false;
+            }
+
+            let cliSemver: semver.SemVer = new semver.SemVer(version);
+            if (cliSemver.compare(ConnectionManager.MINIMAL_JFROG_CLI_VERSION_FOR_DEFAULT_EXPORT) < 0) {
+                this._logManager.logMessage(
+                    'JFrog CLI version is too low to support credentials extraction (needed: ' +
+                        ConnectionManager.MINIMAL_JFROG_CLI_VERSION_FOR_DEFAULT_EXPORT +
+                        ', actual: ' +
+                        version +
+                        ')',
+                    'DEBUG'
+                );
+                return false;
+            }
+        } catch (error) {
+            return false;
+        }
+        return true;
+    }
+
+    private async getJfrogCliVersion(): Promise<string> {
+        const versionPrefix: string = 'jfrog version ';
+        let output: string = execSync('jf -v').toString();
+        if (!output.startsWith(versionPrefix)) {
+            this._logManager.logMessage('Unexpected output to JFrog CLI version command: ' + output, 'DEBUG');
+            return '';
+        }
+        return output.replace(versionPrefix, '').trim();
+    }
+
+    private async getJfrogCliDefaultServerConfiguration(): Promise<boolean> {
+        try {
+            let output: string = execSync('jf c export').toString();
+            let confStr: string = Buffer.from(output, 'base64').toString('ascii');
+            let conf: any = JSON.parse(confStr);
+
+            this._url = conf['url'] || '';
+            this._xrayUrl = conf['xrayUrl'] || '';
+            this._rtUrl = conf['artifactoryUrl'] || '';
+
+            // Get access token if exists without refresh token. Get username & password otherwise.
+            let accessToken: string = conf['accessToken'] || '';
+            let refreshToken: string = conf['refreshToken'] || '';
+            if (accessToken !== '' && refreshToken === '') {
+                this._accessToken = accessToken;
+            } else {
+                this._username = conf['user'] || '';
+                this._password = conf['password'] || '';
+            }
```

Comment:
We'd like to use a password if exist and avoid reading the refresh token, because the refresh token is an internal logic of the CLI.
The logic should be as following: Get password from CLI, if not exists get the access token from CLI.
```suggestion
            // Get username & password if exists. Get access token otherwise.
            let password: string = conf['password'] || '';
            if (password !== '') {
                this._username = conf['user'] || '';
                this._password = password;
            } else {
                this._accessToken = conf['accessToken'] || '';
            }
```

Reply:
We cannot ignore this internal logic because it may result in obtaining an access token with an expiry.
While prioritizing basic auth may fix it,  we decided to follow the same logic as in the CLI that prefers an access token if such exists

Reply:
In addition, prioritizing basic auth will fix it only because we are aware of the internal logic. Once this is logic changes this is will no longer be true.